app/code/Magento/PageCache/Model/App/FrontController/BuiltinPlugin.php

```diff
@@ -89,9 +89,12 @@ public function aroundDispatch(
      */
     protected function addDebugHeaders(ResponseHttp $result)
     {
-        $cacheControl = $result->getHeader('Cache-Control')->getFieldValue();
-        $this->addDebugHeader($result, 'X-Magento-Cache-Control', $cacheControl);
-        $this->addDebugHeader($result, 'X-Magento-Cache-Debug', 'MISS', true);
+        $cacheControlHeader = $result->getHeader('Cache-Control');
+        if ($cacheControlHeader instanceof \Zend\Http\Header\HeaderInterface) {
+            $cacheControl = $cacheControlHeader->getFieldValue();
+            $this->addDebugHeader($result, 'X-Magento-Cache-Control', $cacheControl);
+            $this->addDebugHeader($result, 'X-Magento-Cache-Debug', 'MISS', true);
```

Comment:
I misspoke, I meant X-Magento-Cache-Debug, because it does not rely on the $cacheControl variable. Won't devs still benefit from seeing whether cache was missed? Agree on X-Magento-Cache-Control header